File: ddmlib/src/main/java/com/android/ddmlib/input/InputReaderThread.java
package com.android.ddmlib.input;

import com.android.ddmlib.Log;

/**
 * Created by majipeng on 2017/6/23.
 * <p>
 * 驱动InputReader
 */
public class InputReaderThread extends Thread {
    private static final String TAG = "InputReaderThread";
    private InputReader reader;
    private EventPool inputDispatcher;


    public InputReaderThread(InputReader reader, EventPool inputDispatcher) {
        setName("inputReader-Thread");
        this.reader = reader;
        this.inputDispatcher = inputDispatcher;
    }

    @Override
    public void run() {
        while (!isInterrupted()) {
            MonitorEvent monitorEvent = reader.read();
            if (monitorEvent != null) {
                Log.d(TAG, "read:" + monitorEvent);
                inputDispatcher.dispatchEvent(monitorEvent);
            } else {
                Log.d(TAG, "read null .. exit!!");
                break;
            }
        }
        Log.d(TAG, "finish..");
    }
}

File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputReaderThread.kt
package com.android.ddmlib.input.android

/**
 * Created by majipeng on 2017/6/23.
 */
class InputReaderThread(private val mContext: InputManager) : Thread() {

    private val read = true

    init {
        name = "inputReader-Thread"
    }

    override fun run() {
        while (read) {

            val monitorEvent = mContext.inputReader.readBySync()
            if (monitorEvent != null) {
                mContext.inputDispatcher.dispatchEvent(monitorEvent)
            }
        }
    }
}
